Signal/src/view controllers/MessagesViewController.m

```diff
@@ -184,14 +191,35 @@ - (void)viewDidLoad {
     self.senderId          = ME_MESSAGE_IDENTIFIER;
     self.senderDisplayName = ME_MESSAGE_IDENTIFIER;
 
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(startReadTimer)
-                                                 name:UIApplicationWillEnterForegroundNotification
-                                               object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(cancelReadTimer)
-                                                 name:UIApplicationDidEnterBackgroundNotification
-                                               object:nil];
+
+}
+
+// Observers that should be present iff the view is visible. (e.g. not in setup/teardown).
+- (void)enableVisibleObservers:(BOOL)shouldObserve {
```

Comment:
Maybe use "toggle observers" - I don't think "visible" adds any clarity to what this selector does or why it exists.